docs/aws/audit/shieldmonitoring/rules/guardduty_non_archived_findings.mdx

```diff
@@ -23,6 +23,222 @@ CBP,RBI_MD_ITF,RBI_UCB
+
+2. **List GuardDuty Detectors**: The first step to check if Non Archived Findings are enabled for GuardDuty is to list all the GuardDuty detectors in your AWS account. You can do this by running the following command: `aws guardduty list-detectors`. This command will return a list of detector IDs.
+
+3. **Get Detector Details**: For each detector ID returned in the previous step, you can get the details of the detector by running the following command: `aws guardduty get-detector --detector-id <detector-id>`. Replace `<detector-id>` with the actual detector ID. This command will return the details of the detector including the status of the "Finding Publishing Frequency" which indicates if Non Archived Findings are enabled or not.
+
+4. **Check Finding Publishing Frequency**: In the output of the previous command, look for the "FindingPublishingFrequency" field. If the value of this field is "FIFTEEN_MINUTES", "ONE_HOUR", or "SIX_HOURS", it means that Non Archived Findings are enabled. If the value is "TWENTY_FOUR_HOURS" or "FORTY_EIGHT_HOURS", it means that Non Archived Findings are not enabled.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Import necessary libraries and establish a session**: To start with, you need to import the necessary libraries in your Python script. Boto3 is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, which allows Python developers to write software that makes use of AWS services like Amazon S3, Amazon EC2, etc. Here is how you can do it:
+
+```python
+import boto3
+import json
+
+# Create a session using your AWS credentials
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2' # or any other region where you have your resources
+)
+```
+
+2. **Create a GuardDuty client**: After establishing a session, you need to create a GuardDuty client. This client will allow you to interact with the GuardDuty service and perform various operations.
+
+```python
+# Create GuardDuty client
+gd_client = session.client('guardduty')
+```
+
+3. **List detectors**: GuardDuty uses detectors to monitor the activity in your AWS environment. You need to list all the detectors in your environment to check their configurations.
+
+```python
+# List detectors
+response = gd_client.list_detectors()
+detectors = response['DetectorIds']
+```
+
+4. **Check for non-archived findings**: Now, you can iterate over all the detectors and check if they have non-archived findings enabled. If the `FindingPublishingFrequency` attribute of a detector is set to `FIFTEEN_MINUTES`, `ONE_HOUR`, or `SIX_HOURS`, it means that the detector is configured to automatically archive findings after the specified time period.
+
+```python
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>